src/worker.cc

```diff
@@ -175,22 +175,16 @@ rapidjson::Document from_string(const std::string& json, const valhalla_exceptio
 
 void add_date_to_locations(Options& options,
                            google::protobuf::RepeatedPtrField<valhalla::Location>& locations) {
-  // otherwise we do what the person was asking for
+  // /(trace_)route needs special treatment
   if (options.has_date_time_case() && !locations.empty()) {
     switch (options.date_time_type()) {
       case Options::current:
-        locations.Mutable(0)->set_date_time("current");
-        break;
-      case Options::depart_at:
-        locations.Mutable(0)->set_date_time(options.date_time());
-        break;
-      case Options::arrive_by:
-        locations.Mutable(locations.size() - 1)->set_date_time(options.date_time());
+        for (auto& loc : locations)
+          loc.set_date_time("current");
         break;
-      case Options::invariant:
+      default:
         for (auto& loc : locations)
           loc.set_date_time(options.date_time());
-      default:
         break;
     }
   }
```

Comment:
I don't think it'd matter much for `route` if we set the date_time for all locations, no matter the `date_time.type`, AFAICT it's always implicit by the algorithm how a location's date_time is interpreted. The only one I'm unsure about is `trace_route`